src/cli/pad.rs

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Matteo Dell'Acqua
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+ */
+
+use crate::prelude::*;
+use anyhow::{ensure, Context, Result};
+use clap::{ArgMatches, Args, Command, FromArgMatches, ValueEnum};
+use common_traits::UnsignedInt;
+use log::info;
+use std::{
+    ffi::OsString,
+    mem::size_of,
+    path::{Path, PathBuf},
+};
+
+pub const COMMAND_NAME: &str = "pad";
+
+fn pad(path: impl AsRef<Path>, block_size: u64) -> Result<()> {
+    let file_len = path
+        .as_ref()
+        .metadata()
+        .with_context(|| {
+            format!(
+                "Cannot extract metadata from file {}",
+                path.as_ref().display()
+            )
+        })?
+        .len();
+    let expected_len = file_len.align_to(block_size);
+    if file_len == expected_len {
+        info!(
+            "File {} already aligned to a block size of {} bytes",
+            path.as_ref().display(),
+            block_size
+        );
+    } else {
+        let file = std::fs::File::options()
+            .read(true)
+            .write(true)
+            .open(path.as_ref())
+            .with_context(|| format!("Cannot open file {} to pad", path.as_ref().display()))?;
+        file.set_len(expected_len)
+            .with_context(|| format!("Cannot extend file {}", path.as_ref().display()))?;
+        info!(
+            "File {} successfully zero-padded to align to a block size of {} bytes",
+            path.as_ref().display(),
+            block_size
+        );
+    }
+    Ok(())
+}
+
+#[derive(Args, Debug)]
+#[command(about = "Zero pad graph files to align to the specified size", long_about = None)]
```

Comment:
From what I can understand, you specify a type on the command line, not a size. So it si like "Zero-pad graph files to a length multiple of a read-word type"